src/wp-includes/post.php

```diff
@@ -6369,7 +6369,6 @@ function wp_delete_attachment( $post_id, $force_delete = false ) {
 	 */
 	do_action( 'delete_attachment', $post_id, $post );
 
-	wp_delete_object_term_relationships( $post_id, array( 'category', 'post_tag' ) );
 	wp_delete_object_term_relationships( $post_id, get_object_taxonomies( $post->post_type ) );
```

Comment:
This line was originally added in https://core.trac.wordpress.org/changeset/12305

I'm hesitant to remove the first line without knowing why the second line was needed.

As a next step, I'd suggest writing unit tests covering this scenario that prove the behavior remains the same after the change.

Reply:
> I also think we need to figure out why both lines were included in the first place, and what changed since then that makes the first line no longer relevant.

As it turns out, the answer is in the original Trac ticket! https://core.trac.wordpress.org/ticket/10821#comment:7

We'll need to keep the line, but we can add some better safeguarding around it.